### PR TITLE
Fix contraband available roundstart cola/snacks, add new hacked cola/snack spawner, replace hacked snack and cola vends on ruins

### LINKED ISSUE #56101: Vending machines are being hacked round start
Has been happening consistently on Bagil, vending machines will show their hacked/premium items round start.

### PR DESCRIPTION
**Pull request by kopoba**
## About The Pull Request
Fix https://github.com/tgstation/tgstation/issues/56101
Add new spawner /obj/effect/spawner/randomcolavend/contraband and /obj/effect/spawner/randomsnackvend/contraband
Replace current vends on ruins with new spawner /obj/effect/spawner/randomsnackvend/contraband and /obj/effect/spawner/randomcolavend/contraband

## Why It's Good For The Game
Fix https://github.com/tgstation/tgstation/issues/56101
More tools for mappers remove snowflake var/hacked = TRUE

## Changelog
:cl:
add: New spawner for mappers hacked cola and snacks vendomats
fix: Cola and snacks vendomats not hacked shiftstart
fix: Cola and snacks vendomats on syndicat listnening post and bioweapon base replaced with /obj/effect/spawner/randomsnackvend/contraband
/:cl:

### CODE REVIEW

File: code/game/objects/effects/spawners/vending.dm [hunk 1]
```diff
@@ -1,14 +1,24 @@
...
+
+	return INITIALIZE_HINT_QDEL
+
+/obj/effect/spawner/randomsnackvend/contraband
+	icon = 'icons/obj/vending.dmi'
+	icon_state = "random_snack"
+	name = "Spawn random hacked snack vending machine"
+	desc = "Automagically transforms into a random hacked snack vendor. If you see this while in a shift, please create a bug report."
+
+/obj/effect/spawner/randomsnackvend/contraband/Initialize(mapload)
```
Comment (EgorDinamit) [inline code/game/objects/effects/spawners/vending.dm:21]:
```suggestion
/obj/effect/spawner/randomsnackvend/contraband/Initialize(mapload)
..()
```
You forgot to call parent proc.

File: code/game/objects/effects/spawners/vending.dm [hunk 2]
```diff
@@ -19,16 +29,27 @@
...
+
+	return INITIALIZE_HINT_QDEL
+
+/obj/effect/spawner/randomcolavend/contraband
+	icon = 'icons/obj/vending.dmi'
+	icon_state = "random_cola"
+	name = "Spawn random hacked cola vending machine"
+	desc = "Automagically transforms into a random hacked cola vendor. If you see this while in a shift, please create a bug report."
+
+/obj/effect/spawner/randomcolavend/contraband/Initialize(mapload)
```
Comment (EgorDinamit) [inline code/game/objects/effects/spawners/vending.dm:49]:
```suggestion
/obj/effect/spawner/randomcolavend/contraband/Initialize(mapload)
..()
```

**Comment by EgorDinamit** [comment]
I may be blind, but is "hacked" bool is even used anywhere?

**Comment by EgorDinamit** [comment]
You just committed same suggestion twice.
Remove clone "..()" at line 52.

**Comment by kopoba** [comment]
Sad

**Pull request closed without merging**